workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -110,6 +110,17 @@ namespace <#= qualifiedNamespace #>
                 dirtyBits<#= i #> = 0x0;
 <# } #>
             }
+
+            public Snapshot ToComponentSnapshot(global::Unity.Entities.World world)
+            {
+                var componentDataSchema = new ComponentData(new SchemaComponentData(<#= componentDetails.ComponentId #>));
+                Serialization.SerializeComponent(this, componentDataSchema.SchemaData.Value.GetFields(), world);
+                var snapshot = Serialization.DeserializeSnapshot(componentDataSchema.SchemaData.Value.GetFields(), world);
+
+                componentDataSchema.SchemaData?.Dispose();
```

Comment:
Perhaps null out the SchemaData?

Reply:
Is that necessary? Its not leaking out of this method?

Reply:
It's a good habit to be in, particularly if more generated code is added later that may accidentally access the disposed object.

Reply:
Fair enough. A null ref is better than a segfault I guess!

Reply:
If SchemaData can be null then the line above will throw

Reply:
Line 118 will only throw if there's a bug in the Worker SDK. The constructor sets the `SchemaData` value

Reply:
The `?.` notation was really just shorthand